### ISSUE TITLE
Typing return value duplication

### ISSUE THREAD

**Issue by tbrannam** [shekohex/jsonrpc-ts#16]
When typing the return value like so:

```
    const rpcClient = new RpcClient<MyService>({ url });
    try {
        const {data: {result}} = await rpcClient.makeRequest<'myMethod', ArtistResponse>({ 
            method: 'myMethod', params:  { filter: true }, id: '1', jsonrpc: '2.0' 
        });
        console.log(result);
    } catch (e) {
        console.log(e)
    }
```

It would be nice if there wasn't the redundancy of 'myMethod' in the Typing of template`TRequest` and the `method` function argument.     Is there a point to keeping these separated.  I'm not sure how to express this at the moment

**Comment by shekohex** [comment]
Hi @tbrannam, i just published a new version `0.2.0` which supports the following: 
* Auto Infer return type of a method, if not then fallback to `any`
for example 

```ts
// here is a simple math service
interface MathService {
  // a method called sum that accepts 2 args of type number
  sum: [number, number];
  // or if you need return type, you can have that too :)
  sumWithReturnType: ({ x, y }: { x: number; y: number }) => number;
}
```
take a look at 
```ts
sumWithReturnType: ({ x, y }: { x: number; y: number }) => number;
// -------------------------------------------------------^^^^^^^
```
then the return type of the result response will be infrared to be `number`

```ts
// response.data.result has type of number :)
const response = await rpcClient.makeRequest({
  method: 'sumWithReturnType',
  params: { x: 3, y: 2 },
  id: 2,
  jsonrpc: '2.0',
});
```
also notice that you don't even need to type the method name here in the `mackeRequest<'blahblah'>`
it just got inferred from the `method` key.

> by the way, the old method works too, but instead there is no need anymore for the `TResponse` i removed it completely, because it get inferred to be `any` anyway.

Hope this covers your needs :)

**Comment by tbrannam** [comment]
that's fantastic - nice and clean 👍
